pennylane_lightning/core/src/simulators/lightning_kokkos/measurements/ExpValFunctors.hpp

```diff
@@ -563,38 +566,38 @@ template <class PrecisionT> struct getExpVal5QubitOpFunctor {
     void operator()(const std::size_t k, PrecisionT &expval) const {
         const std::size_t kdim = k * dim;
 
-        INDEX(i00000, 0B00000);
-        INDEX(i00001, 0B00001);
-        INDEX(i00010, 0B00010);
-        INDEX(i00011, 0B00011);
-        INDEX(i00100, 0B00100);
-        INDEX(i00101, 0B00101);
-        INDEX(i00110, 0B00110);
-        INDEX(i00111, 0B00111);
-        INDEX(i01000, 0B01000);
-        INDEX(i01001, 0B01001);
-        INDEX(i01010, 0B01010);
-        INDEX(i01011, 0B01011);
-        INDEX(i01100, 0B01100);
-        INDEX(i01101, 0B01101);
-        INDEX(i01110, 0B01110);
-        INDEX(i01111, 0B01111);
-        INDEX(i10000, 0B10000);
-        INDEX(i10001, 0B10001);
-        INDEX(i10010, 0B10010);
-        INDEX(i10011, 0B10011);
-        INDEX(i10100, 0B10100);
-        INDEX(i10101, 0B10101);
-        INDEX(i10110, 0B10110);
-        INDEX(i10111, 0B10111);
-        INDEX(i11000, 0B11000);
-        INDEX(i11001, 0B11001);
-        INDEX(i11010, 0B11010);
-        INDEX(i11011, 0B11011);
-        INDEX(i11100, 0B11100);
-        INDEX(i11101, 0B11101);
-        INDEX(i11110, 0B11110);
-        INDEX(i11111, 0B11111);
+        std::size_t i00000 = INDEX(i00000, 0B00000);
+        std::size_t i00001 = INDEX(i00001, 0B00001);
+        std::size_t i00010 = INDEX(i00010, 0B00010);
+        std::size_t i00011 = INDEX(i00011, 0B00011);
+        std::size_t i00100 = INDEX(i00100, 0B00100);
+        std::size_t i00101 = INDEX(i00101, 0B00101);
+        std::size_t i00110 = INDEX(i00110, 0B00110);
+        std::size_t i00111 = INDEX(i00111, 0B00111);
+        std::size_t i01000 = INDEX(i01000, 0B01000);
+        std::size_t i01001 = INDEX(i01001, 0B01001);
+        std::size_t i01010 = INDEX(i01010, 0B01010);
+        std::size_t i01011 = INDEX(i01011, 0B01011);
+        std::size_t i01100 = INDEX(i01100, 0B01100);
+        std::size_t i01101 = INDEX(i01101, 0B01101);
+        std::size_t i01110 = INDEX(i01110, 0B01110);
+        std::size_t i01111 = INDEX(i01111, 0B01111);
+        std::size_t i10000 = INDEX(i10000, 0B10000);
+        std::size_t i10001 = INDEX(i10001, 0B10001);
+        std::size_t i10010 = INDEX(i10010, 0B10010);
+        std::size_t i10011 = INDEX(i10011, 0B10011);
+        std::size_t i10100 = INDEX(i10100, 0B10100);
+        std::size_t i10101 = INDEX(i10101, 0B10101);
+        std::size_t i10110 = INDEX(i10110, 0B10110);
+        std::size_t i10111 = INDEX(i10111, 0B10111);
+        std::size_t i11000 = INDEX(i11000, 0B11000);
+        std::size_t i11001 = INDEX(i11001, 0B11001);
+        std::size_t i11010 = INDEX(i11010, 0B11010);
+        std::size_t i11011 = INDEX(i11011, 0B11011);
+        std::size_t i11100 = INDEX(i11100, 0B11100);
+        std::size_t i11101 = INDEX(i11101, 0B11101);
+        std::size_t i11110 = INDEX(i11110, 0B11110);
+        std::size_t i11111 = INDEX(i11111, 0B11111);
```

Comment:
If you could go with a loop here and store the elements in an array, we could greatly reduce code here, and below.

Reply:
The above vectorises nicely though using GCC 11, from avx all the way to avx512f. I'd like to see if using a loop would give the same output before suggesting the change here.